test/jit/test_module_interface.py

```diff
@@ -451,8 +451,150 @@ def forward(self, x):
+                return self.b
+            @torch.jit.export
+            def getb(self, x):
+                return self.b
+
+        class OrigMod(torch.nn.Module):
+            def __init__(self):
+                super(OrigMod, self).__init__()
+                self.a = 0
+
+            def forward(self, x):
+                return self.a
+
+        @torch.jit.interface
+        class ModInterface(torch.nn.Module):
+            def forward(self, x):
+                # type:  (Tensor) -> int
+                pass
+
+        class TestModule(torch.nn.Module):
+            proxy_mod : ModInterface
+
+            def __init__(self):
+                super(TestModule, self).__init__()
+                self.proxy_mod = OrigMod()
+                self.sub = SubModule()
+
+            def forward(self, x):
+                return self.proxy_mod(x) + self.sub.getb(x)
+
+        m = torch.jit.script(TestModule())
+        m.proxy_mod = m.sub
+        m.eval()
+        with self.assertRaisesRegex(RuntimeError, "failed to freeze interface attribute 'proxy_mod'"):
+            mf = torch._C._freeze_module(m._c, freezeInterfaces = True)
+
+    def test_freeze_module_with_interface3(self):
+        class SubModule(torch.nn.Module):
+            def __init__(self):
+                super(SubModule, self).__init__()
+                self.b = torch.tensor([1.5])
+
+            def forward(self, x):
+                self.b[0] += 2;
+                return self.b
+            @torch.jit.export
+            def getb(self, x):
+                return self.b
+
+        class OrigMod(torch.nn.Module):
+            def __init__(self):
+                super(OrigMod, self).__init__()
+                self.a = torch.tensor([0.5])
+
+            def forward(self, x):
+                return self.a
+
+        @torch.jit.interface
+        class ModInterface(torch.nn.Module):
+            def forward(self, x):
+                # type:  (Tensor) -> Tensor
+                pass
+
+        class TestModule(torch.nn.Module):
+            proxy_mod : ModInterface
+
+            def __init__(self):
+                super(TestModule, self).__init__()
+                self.proxy_mod = OrigMod()
+                self.sub = SubModule()
+
+            def forward(self, x):
+                y = self.proxy_mod(x);
+                z= self.sub.getb(x)
+                return y[0] + z[0]
+
+        m = torch.jit.script(TestModule())
+        m.proxy_mod = m.sub
+        m.sub.b = m.proxy_mod.b
+        m.eval()
+        with self.assertRaisesRegex(RuntimeError, "failed to freeze interface attribute 'proxy_mod'"):
+            mf = torch._C._freeze_module(m._c, freezeInterfaces = True)
+
+    def test_freeze_module_with_interface4(self):
+        class SubModule(torch.nn.Module):
+            def __init__(self):
+                super(SubModule, self).__init__()
+                self.b = torch.tensor([1.5])
+
+            def forward(self, x):
+                self.b[0] += 2;
```

Comment:
This is the test case I was talking about, but it's only failing bc `self.b[0] += 2;`, which is an unrelated issue

Reply:
I deleted that line and it still failed. I will upload an update. 

tdlr;  we check attr.type() for mutability. if it is mutable and if moduel.type()->getattribute(name)  then we fail.   In your example self.a has to be preserved, and it is an interface type then it is failed.